README.md

```diff
@@ -217,6 +217,43 @@ variant { Err = variant { BadBurn = record { min_burn_amount = ... } } }
 
 The minting account is also the receiver of the fees burnt in regular transfers.
 
+## Textual representation of accounts
+
+We specify a _canonical textual format_ that all applications should use to display ICRC-1 accounts.
+This format relies on the textual encoding of principals specified in the [Internet Computer Interface Specification](https://internetcomputer.org/docs/current/references/ic-interface-spec/#textual-ids), referred to as `Principal.toText` and `Principal.fromText` below.
+
+### Encoding
+
+Applications SHOULD encode accounts as follows:
+
+  1. The encoding of accounts that do not have a subaccount, such as `{ owner = principal "aaaaa-aa"; subaccount = null }`, is the encoding of the owner principal.
+  2. The encoding of accounts with a non-empty subaccount is the encoding of the concatenation of the owner principal bytes, the subaccount bytes, and an extra byte `A0`<sub>`16`</sub> (160 = 128 + 32).
+
+In pseudocode:
+
+```sml
+encodeAccount({ owner; subaccount = null      }) = Principal.toText(owner)
```

Comment:
what if a subaccount is filled with zeros (`[0;32]`) but not null? I believe that should also be considered null?

Reply:
Yes, it's safe to replace [0;32] with a null in most practical contexts.

However, requiring the optimization in the spec doesn't feel right to me: this approach will break a natural property `forall a : Account . decodeAccount(encodeAccount(a)) == a`.

One benefit of requiring this optimization is that we get a canonical representation (at least until we introduce variable-length subaccounts with padded zeros).

But then, will we fail to explicitly parse textual representations that encode the default accounts? Or "optimize" them as well?

I'd keep the spec as simple as possible, allowing apps to pre-process subaccounts if they want to.

Reply:
That makes sense, but just to double-check for my understanding, the textual representation for an account with explicit subaccount zero is going to look different by the end user than when we do not have a subaccount at all?

Reply:
> the textual representation for an account with explicit subaccount zero is going to look different by the end user than when we do not have a subaccount at all?

Yes, that was the plan.

One problem that comes to mind is potential user confusion when you send to an address and then see a different address in a block explorer because the tooling applied an optimization.

So having a unique encoding might be desirable. But then we need to reject unoptimized encodings in the decoder.